### ISSUE TITLE
false positive: unstable const fn

### ISSUE THREAD

**Issue by Nugine** [rust-lang/rust-clippy#5995]
I tried this code:

```rust
#![deny(clippy::missing_const_for_fn)]

/*
error: this could be a `const fn`
*/

fn is_ascii_digit(byte: u8) -> bool {
    byte.is_ascii_digit()
}

/*
error: `core::num::<impl u8>::is_ascii_digit` is not yet stable as a const fn
*/

// const fn is_ascii_digit(byte: u8) -> bool {
//     byte.is_ascii_digit()
// }

fn main() {
    dbg!(is_ascii_digit(b'1'));
}
```

### Meta

- `cargo clippy -V`: clippy 0.0.212 (04488af 2020-08-24)
- `rustc -Vv`:
  ```
  rustc 1.46.0 (04488afe3 2020-08-24)
  binary: rustc
  commit-hash: 04488afe34512aa4c33566eb16d8c912a3ae04f9
  commit-date: 2020-08-24
  host: x86_64-unknown-linux-gnu
  release: 1.46.0
  LLVM version: 10.0
  ```

**Comment by rail-rain** [comment]
The underlying cause is that `rustc_mir::transform::qualify_min_const_fn::is_min_const_fn`, which is the lint's core, is not responsible to check whether const fns in stable crates call unstable const fns, `rustc_mir::transform::check_consts::validation::Validator` is; but `Validator` panics given a non-const fn. Unless the rework (https://github.com/rust-lang/rust/issues/76618) leaves something useful for the lint, fixing this false positive requires fair amount of rewriting; probably copying some parts of `qualify_min_const_fn`.

**Comment by ecstatic-morse** [comment]
It would be fairly easy to add a "speculative" mode to const-checking. I'll look into this once rust-lang/rust#7661 is complete.
